src/Cargo/sdk/Sdk.targets

```diff
@@ -86,47 +98,71 @@
+    Microsoft.Managed.Targets is imported by the managed language target files in MSBuild 16.0 and
+  above, but most of the msbuild tasks are actually in Microsoft.Common.Currentversion.targets.
     So import it when the managed targets do not get imported.
   -->
-  <Import Project="$(MSBuildToolsPath)\Microsoft.Managed.targets" Condition="'$(MSBuildAssemblyVersion)' >= '16.0' And '$(ManagedLanguageTargetsGotImported)' != 'true'" />
+  <Import Project="$(MSBuildToolsPath)\Microsoft.Managed.targets"
+    Condition="'$(MSBuildAssemblyVersion)' >= '16.0' And '$(ManagedLanguageTargetsGotImported)' != 'true'" />
 
   <!-- Override stock CoreCompile target to do nothing but keep extensibility points -->
   <Target Name="CoreCompile"
-      DependsOnTargets="$(CoreCompileDependsOn)">
-    <CallTarget Targets="$(TargetsTriggeredByCompilation)" Condition="'$(TargetsTriggeredByCompilation)' != ''" />
+    DependsOnTargets="$(CoreCompileDependsOn)">
+    <CallTarget Targets="$(TargetsTriggeredByCompilation)"
+      Condition="'$(TargetsTriggeredByCompilation)' != ''" />
   </Target>
 
   <Target Name="_GenerateCompileInputs" />
   <Target Name="_GenerateCompileDependencyCache" />
 
-  <!-- Disables the _CopyFilesMarkedCopyLocal target to not copy references when SkipCopyFilesMarkedCopyLocal is set to true. -->
-  <Import Project="DisableCopyFilesMarkedCopyLocal.targets" Condition="'$(SkipCopyFilesMarkedCopyLocal)' == 'true'" />
+  <!-- Disables the _CopyFilesMarkedCopyLocal target to not copy references when
+  SkipCopyFilesMarkedCopyLocal is set to true. -->
+  <Import Project="DisableCopyFilesMarkedCopyLocal.targets"
+    Condition="'$(SkipCopyFilesMarkedCopyLocal)' == 'true'" />
   <!-- Rust specific targets -->
-  <UsingTask TaskName="Microsoft.Build.Cargo.CargoTask" AssemblyFile="$(MSBuildThisFileDirectory)..\build\net8.0\Microsoft.Build.Cargo.dll" Condition="'$(MSBuildRuntimeType)' == 'Core' And '$(ShouldImportSdkDll)' != 'false'"/>
-  <UsingTask TaskName="Microsoft.Build.Cargo.CargoTask" AssemblyFile="$(MSBuildThisFileDirectory)..\build\net472\Microsoft.Build.Cargo.dll" Condition="'$(MSBuildRuntimeType)' != 'Core' And '$(ShouldImportSdkDll)' != 'false'"/>
-  <Target Name="CargoFetch" AfterTargets="Restore">
-    <CargoTask EnableAuth="$(AuthMode)" StartupProj="$(MSBuildProjectFullPath)" Command="fetch" />
+  <UsingTask TaskName="Microsoft.Build.Cargo.CargoTask"
+    AssemblyFile="$(MSBuildThisFileDirectory)..\build\net8.0\Microsoft.Build.Cargo.dll"
+    Condition="'$(MSBuildRuntimeType)' == 'Core' And '$(ShouldImportSdkDll)' != 'false'" />
+  <UsingTask TaskName="Microsoft.Build.Cargo.CargoTask"
+    AssemblyFile="$(MSBuildThisFileDirectory)..\build\net472\Microsoft.Build.Cargo.dll"
+    Condition="'$(MSBuildRuntimeType)' != 'Core' And '$(ShouldImportSdkDll)' != 'false'" />
+  <Target Name="CargoInstall" AfterTargets="_CollectRestoreInputs">
```

Comment:
@jeffkl Does this look ok for `AfterTargets` to execute in Restore? Since `AfterTargets="Restore"` won't work in static graph restore. Hate to reference an _ target, but we didn't see another option.

Reply:
Also, would like your input on the `RemoveProperties` if you think we need anything else to de-dupe this.

Reply:
Yeah so `_CollectRestoreInputs` is run for regular command-line restore and Visual Studio design-time build, while `Restore` is run for static graph-based command-line restores as well as regular restores.  But `Restore` is not run in Visual Studio, just `_CollectRestoreInputs`.  Be aware that we changed that logic recently so there's no guarantee that `_CollectRestoreInputs` will continue to work.

Does this need to work in Visual Studio?  If not, I would just run after `Restore`.

> Also, would like your input on the RemoveProperties if you think we need anything else to de-dupe this.

I would hate for a new property to get introduced that breaks this, I wish there was a `RemoveProperties="*"`.  Maybe @rainersigwald has a suggestion to future-proof this.

Reply:
Nothing for you today :( but did finally file it: https://github.com/dotnet/msbuild/issues/11390

Reply:
Thanks for taking a look and opening an issue!